Firmware/src/network/KnomiWebServer.cpp

```diff
@@ -176,6 +176,44 @@ KnomiWebServer::KnomiWebServer(Config *config, WifiManager *manager) {
     req->send(pResponse);
   });
 
+  pServer->on("/api/themeConfig", HTTP_GET, [&](AsyncWebServerRequest *req) {
+    if (this->config != nullptr && this->config->getUiConfig() != nullptr) {
+      AsyncResponseStream *response = req->beginResponseStream("application/json");
+      DynamicJsonDocument doc(128);
+      doc["accentColor"] = String("#") + String(this->config->getUiConfig()->getAccentColor(), HEX);
+
+      serializeJsonPretty(doc, *response);
+      req->send(response);
+    } else {
+      req->send(500, "text/html", "Failed to access UI config");
+    }
+  });
+
+  pServer->on("/api/themeConfig", HTTP_POST, [&](AsyncWebServerRequest *req) {
+    String newAccentColor;
+    if (req->hasArg("accentColor")) {
+      newAccentColor = req->arg("accentColor");
+
+      if (newAccentColor.startsWith("#")) {
+        newAccentColor = newAccentColor.substring(1);
+      }
+    }
+
+    if (this->config != nullptr && this->config->getUiConfig() != nullptr) {
+      if (newAccentColor != nullptr) {
+        LV_LOG_INFO((String("Updating accentColor to: ") + newAccentColor).c_str());
+        this->config->getUiConfig()->setAccentColor(strtol(newAccentColor.c_str(), NULL, 16));
+      }
+      this->config->getUiConfig()->save();
+      req->send(200, "application/json", "{result: \"ok\"}");
+
+      delay(100);
+      ESP.restart();
```

Comment:
Without invalidating objects this was the easiest way to apply the changes, happy to look at invalidating objects instead if you'd prefer?